src/sentry/static/sentry/app/views/performance/transactionSummary/baselineQuery.tsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import isEqual from 'lodash/isEqual';
+
+import {Client} from 'app/api';
+import withApi from 'app/utils/withApi';
+import EventView from 'app/utils/discover/eventView';
+
+export type BaselineQueryResults = {
+  'transaction.duration': number;
+  // event id of the transaction chosen to be the baseline transaction
+  id: string;
+  project: string;
+};
+
+type ChildrenProps = {
+  isLoading: boolean;
+  error: null | string;
+  results: null | BaselineQueryResults;
+};
+
+type Props = {
+  api: Client;
+  eventView: EventView;
+  orgSlug: string;
+
+  children: (props: ChildrenProps) => React.ReactNode;
+};
+
+type State = {
+  fetchID: symbol | undefined;
+} & ChildrenProps;
+
+class BaselineQuery extends React.PureComponent<Props> {
+  state: State = {
+    isLoading: true,
+    fetchID: undefined,
+    error: null,
+
+    results: null,
+  };
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    // Reload data if we aren't already loading,
+    const refetchCondition = !this.state.isLoading && this.shouldRefetchData(prevProps);
+
+    if (refetchCondition) {
+      this.fetchData();
+    }
+  }
+
+  generatePayload(eventView: EventView) {
+    return {
+      ...eventView.getGlobalSelection(),
```

Comment:
I think you want `eventView.getGlobalSelectionQuery()` here.

Reply:
should be fixed in https://github.com/getsentry/sentry/pull/18610/commits/eff6d6b7f39b0c2441d8de362c2f5b45e3a6229e